docs/remote_inference_blueprints/ollama_connector_chat_blueprint.md

```diff
@@ -0,0 +1,193 @@
+# Ollama (OpenAI compatible) connector blueprint example for chat
+
+This is an AI connector blueprint for Ollama or any other local/self-hosted LLM as long as it is OpenAI compatible (Ollama, llama.cpp, vLLM, etc)
+
+## 1. Add connector endpoint to trusted URLs
+
+Adjust the Regex to your local IP.
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://127\\.0\\.0/.*$"
+        ]
+    }
+}
+```
+
+## 2. Enable private addresses
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+      "plugins.ml_commons.connector.private_ip_enabled": true
+    }
+}
+```
+
+## 3. Create the connector
+
+In a local setting, `openAI_key` might not be needed. In case you can either set it to something irrelevant, or if removed, you need to update the `Authorization` header in the `actions`.
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "<YOUR CONNECTOR NAME>",
+  "description": "<YOUR CONNECTOR DESCRIPTION>",
+  "version": "<YOUR CONNECTOR VERSION>",
+  "protocol": "http",
+  "parameters": {
+    "endpoint": "127.0.0.1:11434",
+    "model": "qwen3:4b"
+  },
+  "credential": {
+    "openAI_key": "<YOUR API KEY HERE IF NEEDED>"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://${parameters.endpoint}/v1/chat/completions",
+      "headers": {
+        "Authorization": "Bearer ${credential.openAI_key}"
+      },
+      "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages} }"
+    }
+  ]
+}
+```
+
+### Sample response
+
+```json
+{
+  "connector_id": "Keq5FpkB72uHgF272LWj"
+}
+```
+
+## 4. Register the model
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "Local LLM Model",
+  "function_name": "remote",
+  "description": "Ollama model",
+  "connector_id": "Keq5FpkB72uHgF272LWj"
+}
+```
+
+### Sample response
+
+Take note of the `model_id`. It is going to be needed going forward.
+
+```json
+{
+  "task_id": "oEdPqZQBQwAL8-GOCJbw",
+  "status": "CREATED",
+  "model_id": "oUdPqZQBQwAL8-GOCZYL"
+}
+```
+
+## 5. Deploy the model
+
+Use `model_id` in place of the `<MODEL_ID>` placeholder.
+
+```json
+POST /_plugins/_ml/models/<MODEL_ID>/_deploy
+```
+
+### Sample response
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_deploy
+{
+    "node_ids": ["4PLK7KJWReyX0oWKnBA8nA"]
```

Comment:
To use this, you need to enable `plugins.ml_commons.allow_custom_deployment_plan` setting.
To make the doc easier to follow, I think you can just use `POST /_plugins/_ml/models/<MODEL_ID>/_deploy`